src/components/Carousel/Carousel.ts

```diff
@@ -641,46 +637,86 @@ export const Carousel = defineComponent({
     })
 
     const trackHeight = computed(() => {
-      if (isVertical.value && slideSize.value && config.height === 'auto') {
-        return `${slideSize.value * config.itemsToShow + totalGap.value}px`
+      // If the carousel is vertical and height is set to auto, calculate the height based on slide size and gap
+      if (config.height === 'auto') {
+        if (isVertical.value && slideSize.value) {
+          return `${slideSize.value * config.itemsToShow + totalGap.value}px`
+        }
+        return undefined
+      }
+
+      if (
+        typeof config.height === 'number' ||
+        parseFloat(config.height).toString() === config.height
+      ) {
+        return `${config.height}px`
+      } else {
+        return config.height
       }
-      return config.height !== 'auto'
-        ? typeof config.height === 'number' ||
-          parseInt(config.height).toString() === config.height
-          ? `${config.height}px`
-          : config.height
-        : undefined
     })
 
-    /**
-     * Track style
-     */
-    const trackTransform: ComputedRef<string> = computed(() => {
-      // Calculate the scrolled index with wrapping offset if applicable
-      const cloneOffset = config.wrapAround && mounted.value ? clonedSlidesCount.value : 0
+    const clonedSlidesCount = computed(() => {
+      if (!config.wrapAround) {
+        return { before: 0, after: 0 }
+      }
 
-      // Determine direction multiplier for orientation
-      const directionMultiplier = isReversed.value ? -1 : 1
+      const slidesToClone = Math.ceil(config.itemsToShow)
+      const before = Math.min(slidesToClone, activeSlideIndex.value)
+      const after = Math.min(
+        slidesToClone,
+        slidesCount.value - activeSlideIndex.value - 1
```

Comment:
This calculation for while dragging does not seem correct, if you have 10 slides and are on slide 9 (the last slide, you get after = 0) and it should be the other way around, when you arrive close to the end it should start from 0 and start increasing (it should also include the offset)

I think it should look something more like activeSlideIndex.value + ceil((offset + itemsToShow) / 2) % itemsToShow - slidesCount,

Reply:
Thank you for pointing this out. Indeed, the equation should have been:

```
slidesToClone - (slidesCount.value - activeSlideIndex.value - 1)
```

Let’s consider an example where slidesToClone = 2 and slidesCount = 10. Below is the calculation for different values of activeSlideIndex ranging from 8 to 11:

|activeSlideIndex |	Calculation |	Result|
| --| --|--|
|8	|2 - (10 - 8 - 1) = 2 - (1)|	1|
|9	|2 - (10 - 9 - 1) = 2 - (0)	|2|
|10|	2 - (10 - 10 - 1) = 2 - (-1)	|3|
|11|	2 - (10 - 11 - 1) = 2 - (-2)|	4|